sklearn/metrics/pairwise.py

```diff
@@ -860,6 +860,41 @@ def paired_euclidean_distances(X, Y):
     return row_norms(X - Y)
 
 
+def paired_haversine_distances(X, Y):
+    """
+    Computes the paired Haversine distances between X and Y
+
+    Read more in the :ref:`User Guide <metrics>`.
+
+    Parameters
+    ----------
+    X : array-like, shape (n_samples, 2)
+
+    Y : array-like, shape (n_samples, 2)
+
+    Returns
+    -------
+    distances : ndarray (n_samples, )
+    """
+    X, Y = check_paired_arrays(X, Y)
+    if len(X.shape) != 2 and len(Y.shape) != 2:
+        raise ValueError("Incompatible dimension for X and Y matrices: "
+                         "X.shape and Y.shape should be of to 2 dimension")
+    if not (X.shape[1] == 2 and Y.shape[1] == 2):
+        raise ValueError("Haversine distance only valid in 2 dimensions")
+
+    X, Y = np.radians(X), np.radians(Y)
+    delta = Y-X
+    lat1, lat2 = X[:, 1], Y[:, 1]
+    dlon, dlat = delta[:, 0], delta[:, 1]
+
+    a = np.sin(dlat / 2.0) ** 2 + np.cos(lat1) * np.cos(lat2) * \
```

Comment:
Please use parentheses rather than backslash for line continuation: enclose the whole right hand side expression

Reply:
Done